crates/store/src/server/mod.rs

```diff
@@ -54,18 +63,33 @@ impl Store {
 
         info!(target: COMPONENT, "Database loaded");
 
-        Ok(Self { api_service, listener })
+        Ok(Self {
+            api_service,
+            db_maintenance_service,
+            listener,
+        })
     }
 
     /// Serves the store's RPC API.
     ///
     /// Note: this blocks until the server dies.
     pub async fn serve(self) -> Result<(), ApiError> {
-        tonic::transport::Server::builder()
+        let db_maintenance_service = self.db_maintenance_service.run();
+        let api_service = tonic::transport::Server::builder()
             .trace_fn(miden_node_utils::tracing::grpc::store_trace_fn)
             .add_service(self.api_service)
-            .serve_with_incoming(TcpListenerStream::new(self.listener))
-            .await
-            .map_err(ApiError::ApiServeFailed)
+            .serve_with_incoming(TcpListenerStream::new(self.listener));
+
+        tokio::select! {
+            api_service = api_service => {
+                api_service.map_err(ApiError::ApiServeFailed)
+            },
+
+            _ = db_maintenance_service => {
+                info!(target: COMPONENT, "Database maintenance service shutdown");
+
+                Ok(())
```

Comment:
Are you sure this works as expected?

To me this means the maintenance service will block the api service while its running as only one of these can make progress at once?

I would expect the maintenance service to be `spawn`.